agents/merge_request_agent.py

```diff
@@ -0,0 +1,378 @@
+from tools.filesystem import GetCWDTool, RemoveTool
+from tools.text import (
+    CreateTool,
+    InsertAfterSubstringTool,
+    InsertTool,
+    StrReplaceTool,
+    ViewTool,
+    SearchTextTool,
+)
+from utils import get_agent_execution_config, get_chat_model, get_tool_call_checker_config, mcp_tools, render_prompt
+
+logger = logging.getLogger(__name__)
+
+
+def get_instructions() -> str:
+    return """
+      You are an expert on maintaining packages in RHEL ecosystem. Your job is to tweak existing merge requests
+      and accomodate user feedback.
+
+      To process and accomodate feedback given on a merge request, knowing the target package <PACKAGE>
+      and dist-git branch <DIST_GIT_BRANCH>, do the following:
+
+      1. Go through the comments, including replies if relevant, and follow the provided feedback.
+
+      2. If you updated the spec file, use `rpmlint <PACKAGE>.spec` to validate your changes and fix any new issues.
+
+      3. Verify any changes to patches by running `centpkg --name=<PACKAGE> --namespace=rpms --release=<DIST_GIT_BRANCH> prep`.
+         Repeat as necessary. Do not remove any patches unless all their hunks have been already applied
+         to the upstream sources.
+
+      4. If you removed any patch file references from the spec file (e.g. because they were already applied upstream),
+         you must remove all the corresponding patch files from the repository as well.
+
+      5. Generate a SRPM using `centpkg --name=<PACKAGE> --namespace=rpms --release=<DIST_GIT_BRANCH> srpm`.
+
+      6. In your output, provide a "files_to_git_add" list containing all files that have been modified, added or removed.
+         This typically includes the updated spec file and any new/modified/deleted patch files or other files you've changed
+         or added/removed during processing the feedback. Make sure to include patch files that were also removed
+         from the spec file.
+
+
+      General instructions:
+
+      - If necessary, you can run `git checkout -- <FILE>` to revert any changes done to <FILE>.
+      - Never change anything in the spec file changelog.
+      - Preserve existing formatting and style conventions in spec files and patch headers.
+      - Prefer native tools, if available, the `run_shell_command` tool should be the last resort.
+      - If there are package-specific instructions, incorporate them into your work.
+    """
+
+
+def get_prompt() -> str:
+    return """
+      Your working directory is {{local_clone}}, a clone of source repository of merge request {{merge_request_url}}
+      opened against {{dist_git_branch}} dist-git branch of package {{package}}. The merge request is titled
+      "{{merge_request_title}}" and its description is:
+
+      {{merge_request_description}}
+
+      The merge request contains the following comments with user feedback:
+
+      {{comments}}
+
+      You are working on Jira issue {{jira_issue}}.
```

Comment:
I am wondering if it would be useful giving more context into "what is being fixed", but probablly the `merge_request_description` is usually descriptive enough?